test/Transforms/unroll-jam.mlir

```diff
@@ -7,43 +7,46 @@
 
 // CHECK-LABEL: func @unroll_jam_imperfect_nest() {
 func @unroll_jam_imperfect_nest() {
-  // CHECK: %c100 = constant 100 : index
-  // CHECK-NEXT: affine.for %arg0 = 0 to 100 step 2 {
   affine.for %i = 0 to 101 {
-    // CHECK: "addi32"(%arg0, %arg0) : (index, index) -> i32
-    // CHECK-NEXT: %3 = affine.apply [[MAP_PLUS_1]](%arg0)
-    // CHECK-NEXT: "addi32"(%3, %3) : (index, index) -> i32
     %x = "addi32"(%i, %i) : (index, index) -> i32
     affine.for %j = 0 to 17 {
-      // CHECK:      %8 = "addi32"(%arg0, %arg0) : (index, index) -> i32
-      // CHECK-NEXT: "addi32"(%8, %8) : (i32, i32) -> i32
-      // CHECK-NEXT: %10 = affine.apply [[MAP_PLUS_1]](%arg0)
-      // CHECK-NEXT: %11 = "addi32"(%10, %10) : (index, index) -> i32
-      // CHECK-NEXT: "addi32"(%11, %11) : (i32, i32) -> i32
       %y = "addi32"(%i, %i) : (index, index) -> i32
       %z = "addi32"(%y, %y) : (i32, i32) -> i32
     }
-    // CHECK: "addi32"(%arg0, %arg0) : (index, index) -> i32
-    // CHECK-NEXT: %6 = affine.apply [[MAP_PLUS_1]](%arg0)
-    // CHECK-NEXT: "addi32"(%6, %6) : (index, index) -> i32
-    %w = "addi32"(%i, %i) : (index, index) -> i32
-  } // CHECK }
-  // cleanup loop (single iteration)
-  // CHECK: "addi32"(%c100, %c100) : (index, index) -> i32
-  // CHECK-NEXT: affine.for %arg0 = 0 to 17 {
-  // CHECK-NEXT:   %2 = "addi32"(%c100, %c100) : (index, index) -> i32
-  // CHECK-NEXT:   "addi32"(%2, %2) : (i32, i32) -> i32
-  // CHECK-NEXT: }
-  // CHECK-NEXT: "addi32"(%c100, %c100) : (index, index) -> i32
+    %w = "foo"(%i, %x) : (index, i32) -> i32
+  }
   return
 }
+// CHECK:      affine.for %arg0 = 0 to 100 step 2 {
```

Comment:
Can you fixup this test to avoid hard coding SSA value names?

Reply:
Done. I've replaced similar hard code IVs and block arguments at other places, but there isn't apparently a way to define variables in CHECK-LABEL's -- as a result, block arguments had to be replaced with generic patterns and the test cases aren't as strong now. And for IV definitions, I couldn't avoid hardcoding the prefix "arg", i.e., 

for [[IV0:%arg[0-9]+]]

since

"for [[IV0:%{{.*}}]]"   doesn't appear to match.